veps/sig-compute/node-pressure-eviction.md

```diff
@@ -0,0 +1,229 @@
+# Overview
+
+Kubernetes has two types of pod eviction.
+
+[API-Initiated Eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/) and [Node Pressure Eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/). The API-Initiated Eviction is coordinated at the cluster level. This is the kind of eviction used during node draining events and it’s the type of eviction that observes Pod Disruption Budgets.
+
+Node Pressure Eviction is completely independent from the API-Initiated Eviction. This eviction occurs directly on the node driven by the kubelet. Pod Disruption Budgets are not observed and neither is the pod’s termination grace period. This means when a node is under pressure, the kubelet has the ability to kill workloads independently of any cluster wide rules in order to free up resources.
+
+There are ways to influence and even disable Node Pressure Eviction using the kubelet config, but be aware that these are node level configurations and not directly influenced by the workload itself. **Meaning, unlike API-Initiated Eviction, workload owners cannot directly influence how Node Pressure Eviction behaves for their workload.**
+
+## Motivation
+
+Cluster admins often have two initiatives that conflict with one another. On one hand they want to maximise VM workload density within their cluster to reduce cost. On the other hand they want to minimize disruption to those workloads. Having the ability to automate relieving node pressure by live migrating VMs helps with both of these initiatives.
+
+## Goals
+Currently, the VMI api’s EvictionStrategy tunable only works with API-Initiated Eviction. The goal is to extend the behavior of the EvictionStrategy to Node Pressure Eviction as well. This means if a VMI has an EvictionStrategy of LiveMigrate, we expect the system to attempt to live migrate the VMI regardless of which eviction method initiated the eviction.
```

Comment:
We do integrate with the de-scheduler and to me it seems that it could be used for what the Node Pressure Eviciton is designed. Why would we rather prefer or advertise less reliable approach?

Reply:
@xpivarc can you please elaborate a bit on this idea? How would a descheduler help here?
IIUC, the descheduler is using API-initiated evictions in order to evict workloads and does so in a centralized way. Node-pressure evictions bypass the API completely, being directly triggered by kubelet.

Reply:
>We do integrate with the de-scheduler and to me it seems that it could be used for what the Node Pressure Eviciton is designed

de-scheduler uses api initiated eviction. this is different than node pressure eviction (what we're tryign to solve for here)